src/main/java/it/gov/pagopa/payhub/auth/service/a2a/ClientServiceImpl.java

```diff
@@ -25,4 +35,19 @@ public ClientDTO registerClient(String clientName, String organizationIpaCode) {
 		Client client = clientRegistrationService.registerClient(clientName, organizationIpaCode);
 		return clientMapper.mapToDTO(client);
 	}
+
+	@Override
+	public String getClientSecret(String organizationIpaCode, String clientId) {
+		byte[] clientSecret = clientRetrieverService.getClientSecret(organizationIpaCode, clientId);
+		return dataCipherService.decrypt(clientSecret);
+	}
+
+	@Override
+	public List<ClientDTO> getClients(String organizationIpaCode) {
+		List<Client> clients = clientRetrieverService.getClients(organizationIpaCode);
+		return clients.stream()
+			.map(clientMapper::mapToDTO)
+			.toList();
```

Comment:
I would move even the mapper operation, due to the need to omit the decrypt operation, inside the retrieve Service.

The retriever Service otherwise will just have the logic to call the DAO... which could be less usefull and readable